mongodbatlas/resource_mongodbatlas_maintenance_window.go

```diff
@@ -90,6 +96,13 @@ func resourceMongoDBAtlasMaintenanceWindowCreate(ctx context.Context, d *schema.
 		}
 	}
 
+	if autoDeferValue := d.Get("auto_defer").(bool); autoDeferValue {
+		_, err := conn.MaintenanceWindows.AutoDefer(context.Background(), projectID)
```

Comment:
[nit] `conn.MaintenanceWindows.Defer(ctx, projectID)` -> I see that we are using `ctx` as parameter for defer.

So just to be consistent, could you use `ctx` as parameter to `AutoDefer` instead of `context.Background()`
```suggestion
		_, err := conn.MaintenanceWindows.AutoDefer(ctx, projectID)
```

Reply:
Good catch, wrong copy paste from a previous version i was working on. Fixed